rfcs/text/0000-dss-deletion-daemon.md

```diff
@@ -0,0 +1,631 @@
+              required: false
+              type: string
+            - name: json_request_body
+              in: body
+              required: true
+              schema:
+                type: object
+                properties:
+                  reason:
+                    description: the reason for the deletion.
+                    type: string
+                    enum: [consent_withdrawn, consent_absent, service_disruption, legal]
+                  details
+                    description: User-friendly reason for the bundle or timestamp-specfic bundle deletion.
+                    type: string
+                required:
+                  - reason
+          responses:
+            200:
+              description: Deletion pending
+              schema:
+                type: object
+                  properties:
+                    files:
+                      type: array
+                      description: list of file uuids and versions effected by deletion
+                      items:
+                        type: string
+                    bundles:
+                      type: array
+                      description: list of bundle uuid and versions effected by deletion
+                      items:
+                        type: string
+                    confirmation:
+                      type: string
+                      description: a key used to confirm the deletion operation
+            201:
+              description: Deletion confirmed
+              schema:
+                type: object
+                  properties:
+                    files:
+                      type: array
+                      description: list of file uuids and versions effected by deletion
+                      items:
+                        type: string
+                    bundles:
+                      type: array
+                      description: list of bundle uuid and versions effected by deletion
+                      items:
+                        type: string
+            403:
+              description: Unauthorized user it attempting this action.
+            404:
+              description: the bundle has already been deleted.
+            409:
+              description: The confimation code used is invalid. Either the confirmation code was incorrect, or an effected
+                bundle or file has changed state since the original request.
+```
+    
+A user must have explicit permission to perform a **logical deletion** of a bundle. For a **physical deletion** of a
+bundle, the user must have explicit permission to **physically delete** files and bundles.
+
+(!) The deletion of a bundle does not handle the deletion of secondary analysis bundles referencing this bundle via the 
+links metadata field.
+
+### Restore File API
+
+This is a new endpoint `PUT /restore/file/{uuid}` added to the DSS API which follows this swagger:
```

Comment:
This endpoint seems to have 2 clear use cases from `### Undoing logical deletions`:

1. Restoring a bundle that is still within the 7 day grace period.
2. Restoring a bundle that was permanently deleted.

I'm not sure this swagger seems to have params for each of these cases.  For `2.` you would need to specify where it's getting the file from.

Reply:
Yes, that certainly takes care of it.  Though for case 2, would that just be the normal process of adding a new file or something special?

Reply:
It would be a normal re ingest of the data.